motor/inventory/manager.go

```diff
@@ -232,8 +185,8 @@ func (im *inventoryManager) QuerySecretId(a *asset.Asset) (*vault.Credential, er
 	return im.credentialQueryRunner.Run(a)
 }
 
-func (im *inventoryManager) Resolve(ctx context.Context) map[*asset.Asset]error {
-	resolvedAssets := discovery.ResolveAssets(ctx, im.assetList, im.GetCredential, im.QuerySecretId)
+func (im *inventoryManager) Resolve(ctx context.Context, credsResolver credentials_resolver.Resolver) map[*asset.Asset]error {
```

Comment:
I am wondering if we should set the `credsResover` in `func New(opts ...Option) (*inventoryManager, error)`. By using the option we could always set the default resolver which makes calling the Resolve function easier.

Reply:
done